libraries/AP_Airspeed/AP_Airspeed_SST_ND.cpp

```diff
@@ -0,0 +1,306 @@
+        case DevModel::ND005D:
+            _available_ranges = 6;
+            _range_setting = 3;
+            _current_range_val = nd005d_range[_range_setting];
+            break;
+        default:
+            //GCS_SEND_TEXT(MAV_SEVERITY_ERROR,"ND not setup correctly");
+            return false;
+    }
+
+    // drop to 2 retries for runtime
+    _dev->set_retries(10);
+    
+    _dev->register_periodic_callback(100000, //  6757 for 148Hz ODR 
+                                     FUNCTOR_BIND_MEMBER(&AP_Airspeed_SST_ND::_collect, void));
+    return true;
+}
+
+bool AP_Airspeed_SST_ND::matchModel(uint8_t* reading)
+{ 
+  for (int i = 0; i < 8; i++) {
+    if (reading[i] != MN_SST_ND[i]) {
+      goto probeND210;
+    }
+    _dev_model = DevModel::SST_ND;
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"ND210 dev type detected.\n");
+    return true;
+  }
+  probeND210:
+  for (int i = 0; i < 8; i++) {
+    if (reading[i] != MN_ND210[i]) {
+      goto probeND005;
+    }
+    _dev_model = DevModel::ND210;
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO,"ND210 dev type detected.\n");
+    return true;
+  }
+  probeND005:
+  for (int i = 0; i < 8; i++) {
+    if (reading[i] != MN_ND005D[i]) {
+      return false;
+    }
+    _dev_model = DevModel::ND005D;
+    //GCS_SEND_TEXT(MAV_SEVERITY_INFO, "ND005D dev type detected.\n");
+  }
+  return true;
+}
+
+/*
+    convert raw pressure to pressure in Pascals
+*/
+float AP_Airspeed_SST_ND::_get_pressure(uint32_t dp_raw) const
+{
+    const float margin = 29491.2f;
+    float diff_press_inH2O = 0.0f;
+
+    if (_dev_model != DevModel::SST_ND){
+        diff_press_inH2O = (dp_raw * _current_range_val) / margin;
+    }
+    else{
+        diff_press_inH2O = (float)(_current_range_val *
+                                    (dp_raw - 8388607.5f) /
+                                    15099493.5f);
+    }
+    float press  = diff_press_inH2O * inH20_to_Pa;
+    return press;
+}
+
+/*
+  convert raw temperature to temperature in degrees C
+ */
+float AP_Airspeed_SST_ND::_get_temperature(int8_t dT_int, int8_t dT_frac) const
+{
+    float temp  = dT_int + dT_frac/256.0;
+    return temp;
+}
+
+// read the values from the sensor
+void AP_Airspeed_SST_ND::_collect()
+{
+    uint8_t data[6]; //2 bytes for pressure and 2 for temperature
+    _dev->get_semaphore()->take_blocking();
+    if (!_dev->read(data, sizeof(data))) {
+        return;
+    }
+    _dev->get_semaphore()->give();
+
+    uint32_t dp_raw = 0x00;
+    dp_raw = (data[1] << 16) | (data[2] << 8) | data[3];
+
+    float press  = _get_pressure(dp_raw);
+    float temp  = _get_temperature(data[4], data[5]);
+
+    WITH_SEMAPHORE(sem);
+
+    _press_sum += press ;
+    _temp_sum += temp;
+    _press_count += 1;
+    _temp_count += 1;
+
+    _last_sample_time_ms = AP_HAL::millis();
+}
+
+bool AP_Airspeed_SST_ND::range_change_needed(float last_pressure) {
+    if(last_pressure > 0.8*_current_range_val*inH20_to_Pa){ // if above 85% of range, go to the next
```

Comment:
```suggestion
    if (last_pressure > 0.8*_current_range_val*inH20_to_Pa) { // if above 85% of range, go to the next
```

similarly elsewhere


... and comment is wrong.  Often better to use a generic thing rather than a specific number here...